src/compiler/checker.ts

```diff
@@ -8351,7 +8351,7 @@ namespace ts {
                 // If we have previously computed the control flow type for the reference at
                 // this flow loop junction, return the cached type.
                 const id = getFlowNodeId(flow);
-                const cache = flowLoopCaches[id] || (flowLoopCaches[id] = createMap<Type>());
+                const cache = getOrUpdateArray<Map<Type>>(flowLoopCaches, id, createMap);

```

Comment:
Note that `createMap` takes an argument. If the change recommended below for `getOrUpdateArray` is applied, then it would no longer be safe to pass `createMap` directly.